aws-lc-rs/src/agreement.rs

```diff
@@ -530,25 +575,90 @@ where
         return Err(error_value);
     }
     let peer_pub_bytes = peer_public_key.bytes.as_ref();
-    if peer_pub_bytes.len() != expected_pub_key_len {
-        return Err(error_value);
+    if peer_public_key.encoding == &OCTET_STRING {
+        let peer_pub_bytes_len = peer_pub_bytes.len();
+        if peer_pub_bytes_len != expected_pub_key_len {
```

Comment:
`expected_pub_key_len` here feels confusing if it's OCTET_STRING only. Maybe we can change the method name (pub_key_len) on the Algorithm or move the expected variable declaration into this if?

Reply:
Will move.